src/main/java/technology/dice/dicefairlink/iterators/RandomisedCyclicIterator.java

```diff
@@ -34,6 +35,17 @@ public boolean hasNext() {
     return !elements.isEmpty();
   }
 
+  public boolean hasSameContent(final Collection<T> externalElements) {
+    if (externalElements == null || externalElements.size() != elements.size()) {
```

Comment:
why would `externalElements` ever be null?

Reply:
Because it is an external parameter passed to the public method.

Reply:
Because it is an external parameter passed into the public method.